lib/stripe/request.ex

```diff
@@ -144,6 +144,20 @@ defmodule Stripe.Request do
 
   def get_id!(_), do: raise("You must provide an ID or a struct with an ID to this operation.")
 
+  @spec prefix_expansions(t, binary) :: list
+  def prefix_expansions(%Request{opts: opts} = request, prefix) do
+    case Keyword.get(opts, :expand) do
+      nil ->
+        request
+
+      expansions ->
+        mapped_expansions = Enum.map(expansions, &"#{prefix}.#{&1}")
```

Comment:
What is the need to prefix with `data`?  (sry I prob am not seeing something)

Reply:
The prefix is required when the stripe api is returning a list of resources as they get wrapped in a `data: {}` attribute.

Reply:
@swelham will stripe's response actually expand an expand-able item if the url looks something like this? i.e. prefixed with `data`?

`http://localhost:12123/v1/card?expand[]=data.account`

I thought it has to be like, thus the question of why the `prefix_expansion` is needed.

`http://localhost:12123/v1/card?expand[]=account`

...or for multiple

`http://localhost:12123/v1/card?expand[]=account&expand[]=customer`

I would imagine stripe's response would error...What are your thoughts?

Reply:
In all cases with the exception of returning a list of resources the prefix is not needed. This isn't documented anywhere in the docs as far I can tell and I worked it out by querying the api.

So for create, update and retrieve the following is correct.

`/v1/charges?expand[]=balance_transaction`

Which returns this structure.

```
{
  "id": "ch_1CeIjZC4XIOiOpjmm4mHdB0u",
  "balance_transaction": {
    ...
  },
  ...
}
```

And then for listing resources the following is correct.

`/v1/charges?expand[]=data.balance_transaction`

Which returns this structure.

```
{
  "data": [
    {
      "id": "ch_1CeIjZC4XIOiOpjmm4mHdB0u",
      "balance_transaction": {
        ...
      },
      ...
    }
  ]
}
```

In the case for multiples, yes the format of having multiple `"expand[]="` is correct with the value following the examples above. The stripe api will return a bad request if these are spelt or formatted incorrectly.

Multiples examples.

Api create, update, retrieve

`/v1/charges?expand[]=balance_transaction&expand[]=customer`

Api list.

`/v1/charges?expand[]=data.balance_transaction&expand[]=data.customer`

I hope that makes sense 😃

Reply:
Nice sleuthing!  I just curled the endpoints, and yes, you are completely correct.  💯

Reply:
I'm guessing this also means for `cancel` and `delete`?

Reply:
Yes for `cancel` and no for `delete`. The deletes only return the id of the resource deleted and a flag to indicate success, so there is nothing to expand here.